iocore/net/quic/QUICPacketPayloadProtector_boringssl.cc

```diff
@@ -25,24 +25,114 @@
 #include "QUICPacketPayloadProtector.h"
 #include "tscore/Diags.h"
 
-// static constexpr char tag[] = "quic_ppp";
+static constexpr char tag[] = "quic_ppp";
 
 bool
-QUICPacketPayloadProtector::_protect(uint8_t *protected_payload, size_t &protected_payload_len, size_t max_protecgted_payload_len,
-                                     const Ptr<IOBufferBlock> plain, uint64_t pkt_num, const uint8_t *ad, size_t ad_len,
-                                     const uint8_t *key, const uint8_t *iv, size_t iv_len, const EVP_CIPHER *cipher,
-                                     size_t tag_len) const
+QUICPacketPayloadProtector::_protect(uint8_t *cipher, size_t &cipher_len, size_t max_cipher_len, const Ptr<IOBufferBlock> plain,
+                                     uint64_t pkt_num, const uint8_t *ad, size_t ad_len, const uint8_t *key, const uint8_t *iv,
+                                     size_t iv_len, const EVP_CIPHER *aead, size_t tag_len) const
 {
-  ink_assert(!"not implemented");
-  return false;
+  EVP_CIPHER_CTX *aead_ctx;
+  int len;
+  uint8_t nonce[EVP_MAX_IV_LENGTH] = {0};
+  size_t nonce_len                 = 0;
+
+  this->_gen_nonce(nonce, nonce_len, pkt_num, iv, iv_len);
+
+  if (!(aead_ctx = EVP_CIPHER_CTX_new())) {
+    return false;
+  }
+  if (!EVP_EncryptInit_ex(aead_ctx, aead, nullptr, nullptr, nullptr)) {
+    return false;
```

Comment:
Does aead_ctx need to free ? Can it be a member of QUICTLS ?

Reply:
Good catch. I'll fix this on another PR because the file for OpenSSL has the same issue.

We might be able to make it a member variable, but I don't believe OpenSSL's init functions.
https://github.com/apache/trafficserver/issues/5514